toolchain/check/import_cpp.cpp

```diff
@@ -1677,30 +1691,69 @@ static auto ImportDeclAfterDependencies(Context& context, SemIR::LocId loc_id,
   return SemIR::ErrorInst::InstId;
 }
 
+// Attempts to import a set of declarations. Returns `false` if an error was
+// produced, `true` otherwise.
+static auto ImportDeclSet(Context& context, SemIR::LocId loc_id,
+                          ImportWorklist& decls) -> bool {
+  // Walk the dependency graph in depth-first order, and import declarations
+  // once we've imported all of their dependencies.
+  while (!decls.worklist.empty()) {
+    auto& item = decls.worklist.back();
+    if (!item.added_dependencies) {
+      // We might have already imported this declaration as a dependency of
+      // another declaration in the worklist. If so, don't visit it again.
+      if (IsClangDeclImported(context, item.decl)) {
+        decls.worklist.pop_back();
+        continue;
+      }
```

Comment:
How often would you expect this to happen? Could it end up happening linear-ish number of times?

Mostly wondering if it would be worthwhile to do the check before adding to the worklist rather than here. Ah, but we already do that. Maybe mention in this comment that even though we avoid adding already imported declarations, we can still end up with duplicates where both got added before either was imported?

Reply:
Extended the comment and added an example.